examples/src/main/java/org/apache/spark/examples/streaming/JavaDirectKafkaWordCount.java

```diff
@@ -106,6 +111,32 @@ public Integer call(Integer i1, Integer i2) {
       });
     wordCounts.print();
 
+    //Access the offset ranges using HasOffsetRanges
+    // Hold a reference to the current offset ranges, so it can be used downstream
+    final AtomicReference<OffsetRange[]> offsetRanges = new AtomicReference();
+    messages.transformToPair(
+        new Function<JavaPairRDD<String, String>, JavaPairRDD<String, String>>() {
+            @Override
+            public JavaPairRDD<String, String> call(JavaPairRDD<String, String> rdd) throws Exception {
+            OffsetRange[] offsets = ((HasOffsetRanges) rdd.rdd()).offsetRanges();
+            offsetRanges.set(offsets);
+            return rdd;
+            }
+        }
+    ).foreachRDD(
```

Comment:
I think we could get and print out the offsetRanges immediately in `foreachRDD`, not sure why we need another `transformToPair` to only get the offsetRanges?